src/components/tree/tree.tsx

```diff
@@ -246,94 +252,143 @@ export class Tree {
     const target = event.target as HTMLCalciteTreeItemElement;
 
-    if (root === this.el && target.tagName === "CALCITE-TREE-ITEM" && this.el.contains(target)) {
-      switch (event.key) {
-        case "ArrowDown":
-          const next = target.nextElementSibling as HTMLCalciteTreeItemElement;
-          if (next && next.matches("calcite-tree-item")) {
-            next.focus();
-            event.preventDefault();
-          }
-          break;
-        case "ArrowLeft":
-          // When focus is on an open node, closes the node.
-          if (target.hasChildren && target.expanded) {
-            target.expanded = false;
-            event.preventDefault();
-            break;
-          }
+    if (!(root === this.el && target.tagName === "CALCITE-TREE-ITEM" && this.el.contains(target))) {
+      return;
+    }
 
-          // When focus is on a child node that is also either an end node or a closed node, moves focus to its parent node.
-          const parentItem = target.parentElement.closest("calcite-tree-item");
+    if (event.key === "ArrowDown") {
+      const next = getEnabledSiblingItem(target.nextElementSibling, "down");
+      if (next) {
+        next.focus();
+        event.preventDefault();
+      }
 
-          if (parentItem && (!target.hasChildren || target.expanded === false)) {
-            parentItem.focus();
-            event.preventDefault();
-            break;
-          }
+      return;
+    }
 
-          // When focus is on a root node that is also either an end node or a closed node, does nothing.
-          break;
-        case "ArrowRight":
-          if (!target.hasChildren) {
-            break;
-          }
-          if (target.expanded && getRootNode(this.el).activeElement === target) {
+    if (event.key === "ArrowUp") {
+      const previous = getEnabledSiblingItem(target.previousElementSibling, "up");
+      if (previous) {
+        previous.focus();
+        event.preventDefault();
+      }
+    }
+
+    if (event.key === "ArrowLeft" && !target.disabled) {
+      // When focus is on an open node, closes the node.
+      if (target.hasChildren && target.expanded) {
+        {
+          target.expanded = false;
+          event.preventDefault();
+          return;
+        }
+      }
+
+      // When focus is on a child node that is also either an end node or a closed node, moves focus to its parent node.
+      const parentItem = target.parentElement.closest("calcite-tree-item");
+
+      if (parentItem && (!target.hasChildren || target.expanded === false)) {
+        {
+          parentItem.focus();
+          event.preventDefault();
+          return;
+        }
+      }
+
+      // When focus is on a root node that is also either an end node or a closed node, does nothing.
+      return;
+    }
+
+    if (event.key === "ArrowRight" && !target.disabled) {
+      if (target.hasChildren) {
+        if (target.expanded && getRootNode(this.el).activeElement === target) {
+          {
```

Comment:
Just out of curiosity, what's the purpose of all these extra curly brackets

Reply:
To break the interwebs! 💻🔨

J/K, good catch. I'll remove these unnecessary extra blocks.